### PR TITLE
Issue 5 - "Offline Status" recipe

### PR DESCRIPTION
**Pull request by darkwing**
This recipe covers issue #5.  To Do:

- [x] Cleanup event listeners once we notify the user the app is ready for offline
- [x] Write intro text on `index.html` explaining what this recipe does
- [x] Get rid of gross styling
- [x] Make the example actually do something ?  :D
- [x] Lint

### CODE REVIEW

File: offline-status/index.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+navigator.serviceWorker.register('service-worker.js', {
+  scope: '.'
+}).then(function(registration) {
+  console.log('The service worker has been registered ', registration);
+});
+
+// Listen for claiming of our ServiceWorker
+navigator.serviceWorker.addEventListener('controllerchange', function(event) {
+  console.log('[controllerchange] A "controllerchange" event has happened within navigator.serviceWorker: ', event);
+
```
Comment (darkwing) [inline offline-status/index.js]: Should we do some sort of `scriptURL` validation here to ensure it's the correct service worker?

Comment (digitarald) [inline offline-status/index.js]: We would assume most people have one sw per scope right now. There shouldn't be conflicts.

File: offline-status/service-worker.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+// Files required to make this app work offline
+var REQUIRED_FILES = [
+  'assets/mdn-logo.png',
+  'assets/player.js',
+  'assets/style.css',
+  'assets/users.json',
+  'index.html',
+  'index.js',
+  'service-worker.js'
```
Comment (digitarald) [inline offline-status/service-worker.js]: Not needed.

File: offline-status/index.js (hunk 1 above)
Comment (darkwing) [inline offline-status/index.js]: :+1:

**Comment by darkwing** [comment]
OK, this should be ready for review!

**Comment by darkwing** [comment]
Per @digitarald 's request, I've removed use of the Notification API and instead show a "toaster" notification.

**Comment by digitarald** [comment]
What about showing the notification when the permission was already given, this would be an interesting workaround around randomly showing permission popups without context. The domain could have asked before to show notifications.

**Pull request merged**